### PR TITLE
Fix NodeOptions copy constructor

### PR DESCRIPTION
**Pull request by ivanpauno**

### CODE REVIEW

File: rclcpp/src/rclcpp/node_options.cpp [hunk 1]
```diff
@@ -74,7 +74,10 @@ NodeOptions::operator=(const NodeOptions & other)
     this->enable_rosout_ = other.enable_rosout_;
     this->use_intra_process_comms_ = other.use_intra_process_comms_;
     this->enable_topic_statistics_ = other.enable_topic_statistics_;
+    this->start_parameter_event_publisher_ = other->start_parameter_event_publisher_;
     this->start_parameter_services_ = other.start_parameter_services_;
+    this->parameter_event_qos_ = other.parameter_event_qos_;
+    this->parameter_event_publisher_options_ = other.parameter_event_publisher_options_;
```
Comment (clalancette) [inline rclcpp/src/rclcpp/node_options.cpp:80]: Can we make these the same order they are defined in the class?  That is, `start_parameter_services_`, followed by `start_parameter_event_publisher_`, followed by `parameter_event_qos_`, followed by `rosout_qos_`, followed by `parameter_event_publisher_options_`.  It doesn't matter a whole lot here, but it is just tidier and easier to verify.

**Review by clalancette — CHANGES_REQUESTED** [review]

**Review by clalancette — APPROVED** [review]
Looks good with green CI.

**Comment by ivanpauno** [comment]
@ros-pull-request-builder retest this please
(seems to be flaky tests)

**Comment by ivanpauno** [comment]
CI (only fastrtps, test rclcpp/test_rclcpp):
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12620/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7583/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10327/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12545/)

**Pull request merged**